config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,18 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:default=Internal
+// +kubebuilder:validation:Enum=External;Internal
+type PlatformLoadBalancerType string
+
+const (
+	// LoadBalancerTypeExternal is a load balancer that is managed outside of the cluster.
+	LoadBalancerTypeExternal PlatformLoadBalancerType = "External"
```

Comment:
I would prefer `external` and `internal`  (lowercase) but I saw uppercase being used in ingress types... Any thoughts?